File: python/prograimg_python_basic/Exercises/conditional_statements/conditional_statements_advanced_-_exercise/on_time_for_the_exam.py
exam_hour = int(input())
exam_minute = int(input())
appear_hour = int(input())
appear_minute = int(input())

result = ""
hours = 0
minutes = 0
type_result = ""

exam_hours_in_minutes = (exam_hour * 60) + exam_minute
appear_hour_in_minutes = (appear_hour * 60) + appear_minute

time1 = abs(appear_hour_in_minutes - exam_hours_in_minutes)

if exam_hours_in_minutes < appear_hour_in_minutes:
    print("Late")
    if time1 >= 60:
        hour = time1 // 60
        min = time1 % 60
        if min < 10:
            print(f"{hour}:0{min} hours after the start")
        else:
            print(f"{hour}:{min} hours after the start")
        print(f"{time1} minutes after the start")

    else:
        print(f"{time1} minutes after the start")

elif exam_hours_in_minutes == appear_hour_in_minutes or time1 <= 30:
    print("On time")
    if 1 <= time1 <= 30:
        print(f"{time1} minutes before the start")

else:
    print("Early")
    if time1 >= 60:
        hour = time1 // 60
        min = time1 % 60
        if min < 10:
            print(f"{hour}:0{min} hours before the start")
        else:
            print(f"{hour}:{min} hours before the start")
        print(f"{time1} minutes before the start")
    else:
        print(f"{time1} minutes before the start")

File: JS/JavaScript/JS-Basics/ConditionalStatementsAdvancedExercise/OnTimefortheExam.js
function exam(inputs) {
    let examHourTime = Number(inputs[0]);
    let examMinuteTime = Number(inputs[1]);
    let appearHourTime = Number(inputs[2]);
    let appearMinuteTime = Number(inputs[3]);

    let startExamInMinutes = (examHourTime * 60) + examMinuteTime;
    let appearInMinutes = (appearHourTime * 60) + appearMinuteTime;

    let different =  appearInMinutes - startExamInMinutes;

    if (different < 0){
        let minutes = different * -  1;
        if (minutes <= 30 && minutes < 60) {
            console.log("On time");
            console.log(`${minutes} minutes before the start`);
        }
        else if (minutes > 30 && minutes >= 60) {
            console.log("Early");
            console.log(`${String((minutes / 60))[0]}:${String(minutes % 60).padStart(2, "0")} hours before the start`);
        }
        else{
            console.log("Early");
            console.log(`${minutes} minutes before the start`);
        };

    }

    else if (different > 0){
        console.log("Late");
        if (different < 60) {
            console.log(`${different} minutes after the start`);
        }
        else if (different >= 60) {
            console.log(`${String((different / 60))[0]}:${String(different % 60).padStart(2, "0")} hours after the start`);
        };
    }
    else {
        console.log("On time");
    }

}

exam(["11",
"30",
"8",
"12"])




















